app/src/ui/branches/PublishBranch.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react'
+import { Dispatcher } from '../../lib/dispatcher'
+import { Branch } from '../../models/branch'
+import { ButtonGroup } from '../../ui/lib/button-group'
+import { Button } from '../../ui/lib/button'
+import { Dialog, DialogContent, DialogFooter } from '../../ui/dialog'
+import { Repository } from '../../models/repository'
+
+interface IPublishBranchProps {
+  readonly dispatcher: Dispatcher
+  readonly repository: Repository
+  readonly branch: Branch
+  readonly onPushed: (repository: Repository, branch: Branch) => void
+  readonly onDismissed: () => void
+}
+
+export class PublishBranch extends React.Component<IPublishBranchProps> {
```

Comment:
That was the initial route I was going to take, but I figured I'd get called out on it, so I made two.